FreeRTOS/FreeRTOS
# Unbounded CBMC Proofs for an Array-backed Task-Scheduler

Unbounded CBMC Proofs for an Array-backed Task-Scheduler

Description
-----------
Provide unbounded CBMC proofs (i.e. proofs for task-lists of arbitrary length) for the task-scheduler. However, these proofs assume an array-backed implementation of lists.


Test Steps
-----------
Only the CBMC proofs folder was modified. All CBMC proofs pass.


Related Pull Request
-----------
This pull-request is dependent on the following pull-request which modifies the old linked-list based implementation to an array-list based list implementation

https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/561

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
